File: Data Structures & Algorithms/binary-search/submission-1.java
class Solution {
    public int search(int[] nums, int target) {
        return search(nums, target, nums.length / 2);
    }

    private int search(int[] nums, int target, int index) {
        if (index < 0 || index >= nums.length) {
            return -1;
        } else if (nums[index] == target) {
            return index;
        } else if (nums[index] > target) {
            if (index - 1 >= 0 && nums[index - 1] < target) {
                return -1;
            }
            return search(nums, target, index - 1);
        } else {
            if (index + 1 < nums.length && nums[index + 1] > target) {
                return -1;
            }
            return search(nums, target, index + 1);
        }
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        seq = dict()
        largest = 0
        for n in sorted(nums):
            if n - 1 in seq:
                seq[n] = seq[n - 1] + 1
                del seq[n - 1]
            if n not in seq:
                seq[n] = 1
            if seq[n] > largest:
                largest = seq[n]
        return largest